redfish-core/lib/cable.hpp

```diff
@@ -60,17 +99,236 @@ inline void
             // Cable length is NaN by default, do not throw an error
             if (!std::isnan(*length))
             {
-                messages::internalError(resp);
+                messages::internalError(asyncResp->res);
                 return;
             }
         }
         else
         {
-            resp.jsonValue["LengthMeters"] = *length;
+            asyncResp->res.jsonValue["LengthMeters"] = *length;
         }
     }
 }
 
+/**
+ * @brief Fill cable specific properties.
+ * @param[in,out]   asyncResp       Async HTTP response.
+ * @param[in]       ec          Error code corresponding to Async method call.
+ * @param[in]       properties  List of Cable Properties key/value pairs.
+ */
+inline void
+    fillCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& cableObjectPath,
+                        const std::string& service)
+{
+    sdbusplus::asio::getAllProperties(
+        *crow::connections::systemBus, service, cableObjectPath,
+        "xyz.openbmc_project.Inventory.Item.Cable",
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::DBusPropertiesMap& properties) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
```

Comment:
Should this be BMCWEB_LOG_ERROR()?

Reply:
This is not changed in this PR, but it makes sense and I change  it to ERROR (if it ec!=EBADR)